### PR TITLE
Simplified ux and settings dialog to control feature flags

### PR DESCRIPTION
**Pull request by teresaqhoang**
### Motivation and Context

This PR defaults to a simplified UX experience and allows users more control by adding feature flag support and a settings dialog to control feature visibility.

### Description

- Support for dark mode
- Refactored App component to use FluentProvider and add dark/light theme support.
- Refactored ChatList and ChatListItem components to add action buttons buttons: ListItemActions component to display edit, download, and share actions for chat list items. 
- Added DeleteChatDialog component to delete a chat (Frontend only, feature is disabled by default). 
- Added TabView component to handle tabbed views in the chat. 
- Added UserSettingsMenu component to trigger settings dialog. 
- Introduced SettingsDialog component to control feature visibility, with support for feature descriptions and disabled / inactive features.
image
image
image

Dark Mode
image

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: webapp/src/components/chat/chat-list/ChatListSection.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+export const ChatListSection: React.FC<IChatListSectionProps> = ({ header, conversations }) => {
+    const classes = useClasses();
+    const { selectedId } = useAppSelector((state: RootState) => state.conversations);
+    const keys = Object.keys(conversations);
+
+    return keys.length > 0 ? (
+        <div className={classes.root}>
+            <Text className={classes.header}>{header}</Text>
+            {Object.keys(conversations).map((id) => {
```
Comment (dehoward) [inline webapp/src/components/chat/chat-list/ChatListSection.tsx:44]:
```suggestion
            {keys.map((id) => {
```

File: webapp/src/components/header/UserSettingsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+} from '@fluentui/react-components';
+import { AuthHelper } from '../../libs/auth/AuthHelper';
+import { useAppSelector } from '../../redux/app/hooks';
+import { RootState, resetState } from '../../redux/app/store';
+import { FeatureKeys } from '../../redux/features/app/AppState';
+import { SettingsDialog } from './settings-dialog/SettingsDialog';
+
+export const useClasses = makeStyles({
+    root: {
+        marginRight: '20px',
```
Comment (dehoward) [inline webapp/src/components/header/UserSettingsMenu.tsx:27]:
```suggestion
        marginRight: tokens.spacingHorizontalXL,
```

File: webapp/tests/chat.test.ts [hunk 3]
```diff
@@ -1,51 +1,59 @@
...
+
+        test('Jira', async ({ page }) => {
             test.setTimeout(util.TestTimeout);
-            await plannertests.jiraTest(page) });
-            
-        // Todo: Action Planner intermittently returns a 400 error
+            await plannertests.jiraTest(page);
+        });
+
+        // Todo: [Issue #46] Action Planner intermittently returns a 400 error
```
Comment (dehoward) [inline webapp/tests/chat.test.ts:52]: super nit: "TODO" here and in `utils.ts`

**Review by dehoward — APPROVED** [review]

File: webapp/src/components/chat/chat-list/ChatList.tsx [hunk 4]
```diff
@@ -177,65 +180,56 @@ export const ChatList: FC = () => {
...
-                        <NewBotMenu onFileUpload={() => fileUploaderRef.current?.click()} />
-
-                        <FileUploader
-                            ref={fileUploaderRef}
-                            acceptedExtensions={['.txt', '.json']}
-                            onSelectedFile={onUpload}
-                        />
+                        {!isFiltering && (
+                            <>
+                                <Text weight="bold" size={500} className={classes.title} as="h2">
```
Comment (TaoChenOSU) [inline webapp/src/components/chat/chat-list/ChatList.tsx:189]: nit: consider presets: https://react.fluentui.dev/?path=/docs/components-text--default

File: webapp/src/components/chat/chat-list/dialogs/DeleteChatDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                <Tooltip content={'Delete chat session'} relationship="label">
+                    <Button icon={<Delete16 />} appearance="transparent" aria-label="Edit" />
+                </Tooltip>
+            </DialogTrigger>
+            <DialogSurface className={classes.root}>
+                <DialogBody>
+                    <DialogTitle>Are you sure you want to delete chat {chatName}?</DialogTitle>
+                    <DialogContent>
+                        This will permanently delete the chat for you—but not for Chat Copilot. You need to delete
+                        anything that you have shared (files, tasks, etc.) separately.
```
Comment (TaoChenOSU) [inline webapp/src/components/chat/chat-list/dialogs/DeleteChatDialog.tsx:43]: Is this true?

File: webapp/src/components/chat/shared/Alerts.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    alert: {
+        fontWeight: tokens.fontWeightRegular,
+        color: tokens.colorNeutralForeground1,
+        backgroundColor: tokens.colorNeutralBackground6,
+        fontSize: tokens.fontSizeBase200,
+        lineHeight: tokens.lineHeightBase200,
+    },
+});
+
+export const Alerts: React.FC = () => {
```
Comment (TaoChenOSU) [inline webapp/src/components/chat/shared/Alerts.tsx:26]: We already have an Alerts component under components/shared

File: webapp/src/components/chat/shared/EditChatName.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    chatId: string;
+    exitEdits: () => void;
+    textButtons?: boolean;
+}
+
+export const EditChatName: React.FC<IEditChatNameProps> = ({ name, chatId, exitEdits, textButtons }) => {
+    const classes = useClasses();
+    const dispatch = useAppDispatch();
+    const { instance, inProgress } = useMsal();
+    const chatService = new ChatService(process.env.REACT_APP_BACKEND_URI as string);
```
Comment (TaoChenOSU) [inline webapp/src/components/chat/shared/EditChatName.tsx:43]: Why not use useChat?

File: webapp/src/components/chat/chat-list/dialogs/DeleteChatDialog.tsx (hunk 5 above)
Comment (teresaqhoang) [reply webapp/src/components/chat/chat-list/dialogs/DeleteChatDialog.tsx:43]: Yeah, we'll basically be gravetombing the chat so it's never accessible from the front end, but all the artifacts will be maintained in storage (unless we decide to change it later) 

I also added a TODO to get the right copy from Matthew to communicate this better when we enable deletion

File: webapp/src/components/chat/shared/EditChatName.tsx (hunk 7 above)
Comment (teresaqhoang) [reply webapp/src/components/chat/shared/EditChatName.tsx]: This was copied from the implementation before -- added editChatName in useChat and calling that instead

File: webapp/src/components/chat/shared/Alerts.tsx (hunk 6 above)
Comment (teresaqhoang) [reply webapp/src/components/chat/shared/Alerts.tsx]: Removed

**Review by dehoward — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

**Pull request merged**